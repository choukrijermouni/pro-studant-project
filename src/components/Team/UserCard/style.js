import { blue } from '@pro_boa/ui'
export default {
  row: {
    margin: '16px 0',
    display: 'flex',
    justifyContent: 'flex-start',
    overflow: 'hidden',
    alignItems: 'center',
    padding: 24,
    cursor: 'pointer',
    width: '100%'
  },
  checkBox: {
    width: '0px !important'
  },
  card: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'flex-start',
    width: '10vw'
  },
  licenseCard: {
    display: 'flex',
    alignItems: 'center',
    marginLeft: 24,
    width: ({ invited, scale }) => invited ? scale > 1 ? '22vw' : '23vw' : null
  },
  icon: {
    fontSize: 16,
    color: blue[1]
  },
  flex: {
    height: 120,
    display: 'flex',
    width: '100%',
    transition: 'transform 0.2s ease',
    '&:hover': {
      transform: 'scale(1.01) translateY(-2px)'
    },
    '&:hover $avatarContainer': {
      transform: 'scale(1.15)'
    }
  },
  avatarContainer: {
    transition: 'transform 0.2s ease'
  },
  nameContainer: {
    width: ({ invited, loading }) => (invited && !loading) ? '40vw' : '16vw',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'flex-start',
    marginRight: 25
  },
  skeletonContainer: {
    width: 170
  }
}
