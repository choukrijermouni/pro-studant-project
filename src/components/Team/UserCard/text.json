{
  "expire": "Expire le  ",
  "NV": "Sans licence",
  "never": "Aucune activité",
  "noDate": "Aucune date",
  "creationDate": "Date de création",
  "lastConnectionDate": "Date de dernière connexion",
  "subscriptionExpirationDate": "Date d’expiration de l'abonnement",
  "subscriptionExpired": "Licence expirée",
  "userInvited": "Invité",
  "licenseTypes": {
    "Annual": "Licence annuelle",
    "Monthly": "Licence mensuelle",
    "Trial": "Licence d’essai",
    "None": "Sans licence"
  }
}