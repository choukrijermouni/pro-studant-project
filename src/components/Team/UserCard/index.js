import React from 'react'
import { createUseStyles } from 'react-jss'
import style from './style'
import { Paper, TypographyElement, Avatar, neutral, Icon, Icons, red, Skeleton } from '@pro_boa/ui'
import Text from './text.json'
import moment from 'moment'
import { amazonBucket, Avatars, DMYDateFormat, OrganizationLicenseTypeEnum } from 'constants/'
import { avatarName, isExpired } from 'helpers'

export default ({ name, photo, date, lastConnection, expireDate, handleClick, invited, invitationEmail, invitationLicenseType, organizationLicenseType, loading }) => {
  const scale = window.devicePixelRatio
  const useStyle = createUseStyles(style)
  const {
    row,
    card,
    icon,
    flex,
    avatarContainer,
    licenseCard,
    nameContainer,
    skeletonContainer
  } = useStyle({ invited, scale, loading })
  return (
    <div
      className={flex}
      onClick={handleClick}
      data-test={invited ? 'invited-user-card' : 'learner-card'}
    >
      <Paper className={row}>
        <div className={nameContainer}>
          <div className={avatarContainer}>
            <Avatar
              loading={loading}
              key={loading}
              size='size_1'
              name={avatarName(name)}
              img={invited ? `${amazonBucket.avatar}${Avatars.anonymous}` : photo ? `${amazonBucket.bucketBaseUrl}${photo}` : null}
              dot={invited && !loading}
              dotColor={red[3]}
            />
          </div>
          {
            loading
              ? <Skeleton lines={1} height={20} margin='0px 0px 0px 20px' width={170} />
              : (
                <TypographyElement
                  component='h3'
                  variant='heading3'
                  align='left'
                  fontSize={16}
                  spacing='0px 0px 0px 20px'
                  color={neutral[6]}
                >
                  {invited ? `${Text.userInvited} : ${invitationEmail}` : name || ''}
                </TypographyElement>)

          }
        </div>

        {
          invited && (
            loading
              ? null
              : (
                <div className={licenseCard} title={Text.subscriptionExpirationDate}>
                  <Icon iconName={Icons.licenses} style={icon} />
                  <TypographyElement
                    component='h4'
                    variant='caption1'
                    align='left'
                    spacing='10px 16px 11px 16px'
                    color={neutral[4]}
                  >
                    {
                  invitationLicenseType && invitationLicenseType !== OrganizationLicenseTypeEnum.None
                    ? Text.licenseTypes[invitationLicenseType]
                    : Text.NV
                }
                  </TypographyElement>
                </div>
                )
          )
        }
        {
          !invited
            ? (
              <>
                <div className={card} title={Text.creationDate}>
                  {
                    loading
                      ? <Skeleton lines={1} height={20} margin='10px 16px 11px 16px' width={170} />
                      : (
                        <>
                          <Icon iconName={Icons.calendar} style={icon} />
                          <TypographyElement
                            component='h4'
                            variant='caption1'
                            align='left'
                            spacing='0px 0px 0px 20px'
                            color={neutral[4]}
                          >
                            {date ? moment(date).format(DMYDateFormat) : Text.noDate}
                          </TypographyElement>
                        </>
                        )
                    }
                </div>
                <div className={card} title={Text.lastConnectionDate}>
                  {
                    loading
                      ? <Skeleton lines={1} height={20} margin='0px 0px 0px 20px' width={170} />
                      : (
                        <>
                          <Icon iconName={Icons.lastUserLogin} style={icon} />
                          <TypographyElement
                            component='h4'
                            variant='caption1'
                            align='left'
                            spacing='0px 0px 0px 20px'
                            color={neutral[4]}
                          >
                            {lastConnection ? moment(lastConnection).format(DMYDateFormat) : Text.never}
                          </TypographyElement>
                        </>
                        )
                    }
                </div>

                <div className={licenseCard} title={Text.subscriptionExpirationDate}>
                  {
                    loading
                      ? <div className={skeletonContainer}><Skeleton lines={1} height={20} margin='10px 16px 11px 16px' width={300} /></div>
                      : (
                        <>
                          <Icon iconName={Icons.licenses} style={icon} />
                          <TypographyElement
                            component='h4'
                            variant='caption1'
                            align='left'
                            spacing='0px 0px 0px 20px'
                            color={neutral[4]}
                          >
                            {
                          expireDate
                            ? isExpired(expireDate)
                                ? Text.NV
                                : `${Text.licenseTypes[organizationLicenseType]} ${Text.expire}${moment(expireDate).format(DMYDateFormat)}`
                            : Text.NV
                        }
                          </TypographyElement>
                        </>
                        )
                }
                </div>
              </>
              )
            : loading && (
              <>
                <div className={card} title={Text.creationDate}>
                  <Skeleton lines={1} height={20} margin='10px 16px 11px 16px' width={170} />
                </div>
                <div className={card} title={Text.lastConnectionDate}>
                  <Skeleton lines={1} height={20} margin='0px 0px 0px 20px' width={170} />
                </div>

                <div className={licenseCard} title={Text.subscriptionExpirationDate}>
                  <div className={skeletonContainer}><Skeleton lines={1} height={20} margin='10px 16px 11px 16px' width={300} /></div>
                </div>
              </>
            )
        }
      </Paper>
    </div>
  )
}
