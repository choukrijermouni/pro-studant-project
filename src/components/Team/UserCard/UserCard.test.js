import { shallow, configure } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import renderer from 'react-test-renderer'
import UserCard from '.'
configure({ adapter: new Adapter() })

describe('>>> UserCard --- Shallow Render ManagerCard component', () => {
  let wrapper
  let tree

  beforeEach(() => {
    wrapper = shallow(<UserCard />)
    tree = renderer.create(
      <UserCard />
    ).toJSON()
  })

  it('+++ render correctly and match snapshot', () => {
    expect(tree).toMatchSnapshot()
  })

  it('+++ render the component', () => {
    expect(wrapper).toBeTruthy()
  })
})
