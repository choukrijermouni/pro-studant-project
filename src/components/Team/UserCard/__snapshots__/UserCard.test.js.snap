// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> UserCard --- Shallow Render ManagerCard component +++ render correctly and match snapshot 1`] = `
<div
  className="row-0-2-5 paper-0-2-9"
>
  <div
    className="inputGroup-0-2-10"
  >
    <input
      className="input-0-2-12"
      disabled={false}
      id="desktop1"
      type="checkbox"
    />
    <label
      className="checkbox-0-2-11 checkbox-d0-0-2-14"
    />
    <label
      className="label-0-2-13 checkBox-0-2-6"
      htmlFor="desktop1"
    >
      
    </label>
  </div>
  <div
    className="avatarContainer-0-2-16"
  >
    <div
      className="dot-0-2-15"
    />
    <span
      className="avatarGroup-0-2-17 avatarGroup-d0-0-2-18"
    >
      YS
    </span>
  </div>
  <p
    className="text-0-2-19 text-d0-0-2-20"
  />
  <div
    className="card-0-2-7"
  >
    <span
      className="icon-creation-date-alph-iconfont icon-0-2-8"
    />
    <p
      className="text-0-2-21 text-d0-0-2-22"
    />
  </div>
  <div
    className="card-0-2-7"
  >
    <span
      className="icon-last-login-user-alph-iconfont icon-0-2-8"
    />
    <p
      className="text-0-2-23 text-d0-0-2-24"
    />
  </div>
  <div
    className="card-0-2-7"
  >
    <span
      className="icon-licenses-quantity-alph-iconfont icon-0-2-8"
    />
    <p
      className="text-0-2-25 text-d0-0-2-26"
    >
      Licence annuel expire le  
       
    </p>
  </div>
</div>
`;
